binarytree/__init__.py

```diff
@@ -136,24 +136,22 @@ def _build_tree(values):
     root = _new_node(values[0])
     nodes[0] = root
 
-    index = 1
-    while index < len(values):
-        value = values[index]
-        if value != _null:
-            parent_index = int((index + 1) / 2) - 1
-            parent_node = nodes[parent_index]
-            if parent_node == _null:
-                raise ValueError(
-                    'Node missing at index {}'
-                    .format(parent_index)
-                )
-            child_node = _new_node(value)
-            if index % 2:  # is odd
-                _add_left(parent_node, child_node)
-            else:
-                _add_right(parent_node, child_node)
-            nodes[index] = child_node
-        index += 1
+    for index, value in enumerate(values[1:], 1):
```

Comment:
A slicing operation like `values[1:]` creates a new list, and is costly when the number of elements in the list is large. While your suggested change does make the code cleaner, I don't think it justifies sacrificing performance albeit small.